tests/auth_tests/models/with_integer_username.py

```diff
@@ -0,0 +1,26 @@
+from django.contrib.auth.models import AbstractBaseUser, BaseUserManager
+from django.db import models
+
+
+class CustomUserWithIntegerUsernameManager(BaseUserManager):
+    def create_user(self, username, password):
+        user = self.model(username=username)
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    def get_by_natural_key(self, username):
+        return self.get(username=username)
+
+
+class CustomUserWithIntegerUsername(AbstractBaseUser):
```

Comment:
Nitpick: You may want to add `CustomUserWithIntegerUsername` to `__all__` in `tests/auth_tests/models/__init__.py`.